src/modules/services-systemd/main.py

```diff
@@ -23,66 +24,59 @@
 
 
 def pretty_name():
-    return _("Configure systemd services")
+    return _("Configure systemd units")
 
 
-def systemctl(targets, command, suffix):
+def systemctl(command, units):
     """
-    For each entry in @p targets, run "systemctl <command> <thing>",
-    where <thing> is the entry's name plus the given @p suffix.
-    (No dot is added between name and suffix; suffix may be empty)
+    For each entry in @p units, run "systemctl <command> <thing>",
+    where <thing> is the entry's full name.
 
     Returns a failure message, or None if this was successful.
-    Services that are not mandatory have their failures suppressed
+    Units that are not mandatory have their failures suppressed
     silently.
     """
-    for svc in targets:
-        if isinstance(svc, str):
-            name = svc
+    for unit in units:
+        if isinstance(unit, str):
+            name = unit
             mandatory = False
         else:
-            name = svc["name"]
-            mandatory = svc.get("mandatory", False)
+            name = unit["name"]
+            mandatory = unit.get("mandatory", False)
```

Comment:
Is this still needed or is it legacy code?

Reply:
It handles the two forms of input for specifying units: (1) Just a name, and (2) A name and whether it is mandatory or not. It looks like it is still necessary

Reply:
It looks like it is checking if `unit` is a string.  If it is, it sets the name to it.

That doesn't look like a valid option at this point.  Am I missing something?

Reply:
> 

It could either be a string (just the name) , or a map that contains a name and a flag. Reminding that this PR grouped the units by action, and the systemctl function is called with a list of unit information, just like how it was earlier. This information either contains just the name or a map of name and a boolean flag.